music21/interval.py

```diff
@@ -3490,10 +3493,40 @@ def _diatonicTransposePitch(self,
                 # cannot import Accidental here.
                 pitch2.accidental = halfStepsToFix  # type:ignore
 
-            # inherit accidental display type etc. but not current status
-            if pitch2.accidental is not None and pitch1.accidental is not None:
-                pitch2.accidental.inheritDisplay(pitch1.accidental)
-                pitch2.accidental.displayStatus = None  # set accidental display to None
+            # inherit accidental display options
+            accid1: Accidental
+            if pitch1.accidental is None:
+                # Make a real Accidental that represents "non-displayed natural"
+                accid1 = Accidental('natural')
+                accid1.displayStatus = False  # implied by pitch1.accidental is None
+            else:
+                accid1 = pitch1.accidental
+
+            accid2: Accidental
+            if pitch2.accidental is None:
+                # Make a real Accidental that represents "non-displayed natural"
+                accid2 = Accidental('natural')
+                accid2.displayStatus = False  # implied by pitch2.accidental is None
+            else:
+                accid2 = pitch2.accidental
+
+            accid2.inheritDisplay(accid1)
+
+            if pitch2.accidental is None:
+                # accid2 was a made-up Accidental that represents "non-displayed natural",
```

Comment:
here's where accid2 might be created only if needed

```
if pitch1.accidental is not None:
    pitch2.accidental = 0
    pitch2.accidental.inheritDisplay(pitch1.accidental)
```

Reply:
I rewrote this code to be cleaner (I'll do the "make it optional" thing in a bit).